src/main/java/games/battlelore/BattleloreForwardModel.java

```diff
@@ -0,0 +1,245 @@
+package games.battlelore;
+
+import core.AbstractForwardModel;
+import core.AbstractGameState;
+import core.actions.AbstractAction;
+import core.components.GridBoard;
+
+import core.interfaces.IGamePhase;
+import core.rules.Node;
+import games.battlelore.actions.AttackUnitsAction;
+import games.battlelore.actions.MoveUnitsAction;
+import games.battlelore.actions.PlayCommandCardAction;
+import games.battlelore.actions.SkipTurnAction;
+import games.battlelore.cards.CommandCard;
+import games.battlelore.components.MapTile;
+import games.battlelore.components.Unit;
+
+import utilities.Utils;
+
+import java.util.*;
+
+
+public class BattleloreForwardModel extends AbstractForwardModel {
+    @Override
+    protected void _setup(AbstractGameState initialState) {
+        Random random = new Random(initialState.getGameParameters().getRandomSeed());
+        BattleloreGameParameters gameParams = (BattleloreGameParameters) initialState.getGameParameters();
+        BattleloreGameState gameState = (BattleloreGameState)initialState;
+        BattleloreData _data = gameState.getData();
+
+        for (int i = 0; i  < gameState.playerCount; i++) {
+            if (gameState.playerCount != 2) {
```

Comment:
This doesn't need to be in the for loop, and it should raise an exception, not just a quiet error message.

Reply:
done in part 1